java/steps/yaks-jms/src/test/resources/org/citrusframework/yaks/jms/jms.feature

```diff
@@ -0,0 +1,12 @@
+Feature: JMS steps
+
+  Background:
+    Given connection factory class: org.apache.activemq.ActiveMQConnectionFactory
```

Comment:
How about using a data table with key-value pairs that defines the connection factory. Something like:

```
Given JMS connection factory
  | type      | org.apache.activemq.ActiveMQConnectionFactory
  | brokerUrl | tcp://localhost:61616
```

This would be a bit more readable. Also different connection factory implementations might have different constructor arguments and properties.

Reply:
With the list of arguments, the user has the freedom to specify parameters in the correct order. 

With the key-value map: As a user, I would expect that the key will match with a constructor argument name. This is not possible to achieve because parameter names are lost at compile time.

The only option with key-value map is to use setters instead of the constructor. I think there may be ConnectionFactory implementation which does not support empty constructor.

Reply:
True but it is our job to provide proper connection factory instantiation logic. The user has to know the constructor and provide the correct order of arguments which is pretty error prone in my eyes.

Instead we could map the given properties to constructor arguments and make sure that we support some common connection factories this way. As a start we can only support ActiveMQ and add more connection factories as they are requested by the community.

As a fallback when the connection factory type is unknown to YAKS we could try to add the properties as constructor arguments by the given order regardless of what key the user has given.